### PR TITLE
Fix: Show View Controller memory leak

### PR DESCRIPTION
**Pull request by tomasbabulak**

### CODE REVIEW

File: Sources/FuntastyKit/Architecture/Coordinator.swift [hunk 1]
```diff
@@ -32,8 +32,10 @@ public protocol DefaultCoordinator: Coordinator {
 }
 
 public protocol ShowCoordinator: DefaultCoordinator {
-    var sourceViewController: UIViewController { get }
+    /// sourceViewController should be used as 'weak var', otherwise memory leak will emerge
+    var sourceViewController: UIViewController? { get }
```
Comment (bezoadam) [inline Sources/FuntastyKit/Architecture/Coordinator.swift:36]: Where is `weak` keyword? Shouldn't be here at the start of line?

Comment (tomasbabulak) [reply Sources/FuntastyKit/Architecture/Coordinator.swift:36]: 'weak' cannot be applied to a property declaration in a protocol

Comment (samoilyk) [inline Sources/FuntastyKit/Architecture/Coordinator.swift:36]:
```suggestion
    var sourceViewController: UIViewController! { get }
```

Comment (eRDe33) [reply Sources/FuntastyKit/Architecture/Coordinator.swift]: @bezoadam very good england exhibition here. 👏 ...jinak rekl bych, ze sis nevsiml toho, ze to je protocol, takze tady to weak byt nemuze, ale aby ten `sourceViewController` mohl byt weak v konkretnim Coordinatoru, tak musi byt optional, protoze weak veci musi byt vzdycky optional.

**Review by eRDe33 — DISMISSED** [review]

File: Sources/FuntastyKit/Architecture/Coordinator.swift (hunk 1 above)
Comment (eRDe33) [reply Sources/FuntastyKit/Architecture/Coordinator.swift]: shit a ja si nevsiml, ze to je do FuntastyKitu, proto ta anglictina :loool:

Comment (tomasbabulak) [reply Sources/FuntastyKit/Architecture/Coordinator.swift:36]: As discussed, in this case all protocols should be updated as all projects that are using FuntastyKit coordinators. So for the sake of consistency and compatibility, it must be optional.

**Review by samoilyk — APPROVED** [review]

**Review by mkj-is — APPROVED** [review]

**Comment by mkj-is** [comment]
Closes #72.

**Pull request merged**